File: site-laravel/app/Models/CommandeWeb.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

/**
 * Commande web d'un client (ou panier en cours).
 *
 * Cycle de vie : panier -> payee -> expediee -> livree.
 * Le statut 'panier' represente le panier actif du client (une seule par client).
 * La conversion panier -> commande se fait via CommandeController::valider().
 *
 * @property int         $id
 * @property int         $id_client
 * @property string      $statut            panier|payee|expediee|livree|annulee
 * @property float       $total_ttc
 * @property string|null $adresse_livraison Snapshot de l'adresse au moment de la commande.
 * @property \Carbon\Carbon|null $date_commande
 *
 * @property-read Client $client
 * @property-read \Illuminate\Database\Eloquent\Collection<CommandeWebLigne> $lignes
 */
class CommandeWeb extends Model
{
    protected $table = 'commandes_web';
    public $timestamps = false;

    protected $fillable = [
        'id_client', 'statut', 'total_ttc',
        'adresse_livraison', 'date_commande',
    ];

    protected $casts = [
        'date_commande' => 'datetime',
        'date_creation' => 'datetime',
        'total_ttc'     => 'decimal:2',
    ];

    /** Client proprietaire de cette commande. */
    public function client(): \Illuminate\Database\Eloquent\Relations\BelongsTo
    {
        return $this->belongsTo(Client::class, 'id_client');
    }

    /** Lignes de commande (produit + quantite + prix). */
    public function lignes(): \Illuminate\Database\Eloquent\Relations\HasMany
    {
        return $this->hasMany(CommandeWebLigne::class, 'id_commande');
    }
}

File: app-csharp/CharlesNadejda/CharlesNadejda/Models/CommandeWeb.cs
using System;
using System.Collections.Generic;

namespace CharlesNadejda.Models
{
    /// <summary>
    /// Commande passée via le site e-commerce Laravel.
    /// Lecture seule côté ERP C# — les commandes sont créées par le site web.
    /// </summary>
    public class CommandeWeb
    {
        // ── Constantes de statut ──────────────────────────────────

        /// <summary>Panier en cours de constitution (pas encore validé).</summary>
        public const string StatutPanier  = "panier";
        /// <summary>Commande validée et payée.</summary>
        public const string StatutPayee   = "payee";
        /// <summary>Commande annulée (par le client ou l'admin).</summary>
        public const string StatutAnnulee = "annulee";

        // ── Colonnes DB (table commandes_web) ─────────────────────

        public int       Id                 { get; set; }
        public int       IdClient           { get; set; }
        /// <summary>Statut de la commande : panier, payee, annulee. Voir constantes StatutXxx.</summary>
        public string    Statut             { get; set; }
        public decimal   TotalTtc           { get; set; }
        public string    AdresseLivraison   { get; set; }
        public DateTime? DateCommande       { get; set; }
        public DateTime  DateCreation       { get; set; }

        // ── Propriétés de jointure (chargées par le DAL) ──────────

        /// <summary>Chargé par CommandeWebDAL via JOIN clients.nom.</summary>
        public string NomClient    { get; set; }
        /// <summary>Chargé par CommandeWebDAL via JOIN clients.prenom.</summary>
        public string PrenomClient { get; set; }
        /// <summary>Chargé par CommandeWebDAL via JOIN clients.email.</summary>
        public string EmailClient  { get; set; }
        /// <summary>Chargé par CommandeWebDAL via sous-requête COUNT sur commandes_web_lignes.</summary>
        public int    NbArticles     { get; set; }
        /// <summary>Chargé par CommandeWebDAL via GROUP_CONCAT (ex: "Baguette x11, Pain x2").</summary>
        public string ResumeArticles { get; set; }

        /// <summary>Lignes détaillées — chargées optionnellement par CommandeWebDAL.GetById().</summary>
        public List<CommandeWebLigne> Lignes { get; set; } = new List<CommandeWebLigne>();

        // ── Propriétés calculées ──────────────────────────────────

        /// <summary>Nom complet = Prénom + Nom du client.</summary>
        public string NomCompletClient => $"{PrenomClient} {NomClient}";

        public override string ToString() => $"Commande #{Id} — {NomCompletClient}";
    }
}
